tests/simple/simple_test.go

```diff
@@ -113,30 +130,95 @@ func TestMain(m *testing.M) {
 	os.Exit(m.Run())
 }
 
-func TestSimpleFile(t *testing.T) {
-	// Special case to handle test invocation without args. See TestMain()
-	// early return comment.
-	if rc == nil {
-		return
-	}
-	// Run the flag-configured tests.
-	for _, filename := range flag.Args() {
-		testFile, err := parseSimpleFile(filename)
-		if err != nil {
-			t.Fatalf("Can't parse input file %v: %v", filename, err)
+ func TestSimpleFile(t *testing.T) {
+        // Special case to handle test invocation without args. See TestMain()
+        // early return comment.
+        if rc == nil {
+                return
+        }
+	var skipTests []string
+	firstFlag := true
+	for _, flagVal := range skipFlags {
+		if !strings.Contains(flagVal, "/") {
+			log.Fatal("Unable to parse skip_test flag, specifically for ", flagVal)
+		}
+		fileInd := strings.Index(flagVal, "/")
+		if (flagVal[:fileInd] == ""){
+			log.Fatal("Empty filename")
 		}
-		t.Logf("Running tests in file %v\n", testFile.Name)
-		for _, section := range testFile.Section {
-			t.Logf("Running tests in section %v\n", section.Name)
-			for _, test := range section.Test {
-				desc := fmt.Sprintf("%s/%s/%s", testFile.Name, section.Name, test.Name)
-				t.Run(desc, func(t *testing.T) {
-					err := rc.RunTest(test)
-					if err != nil {
-						t.Fatal(err)
+		allSections := strings.Split(flagVal, ";")
+		filename := flagVal[:fileInd+1]
```

Comment:
Move the initialization of `filename` up one line, then you can inline `allSections` and avoid naming it.

Also, I think this includes the "/" in the filename - you should exclude it for easier matching.

Reply:
I included the / in the filename because it's only used for creating the full path name with section and test name